### PR TITLE
Closes #204 documentation update reference page with dataset categories

### LINKED ISSUE #204: Documentation:  Update reference page with dataset categories
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

While reviewing a PR it occurred to me that the reference page could be updated to have subcategories for each TA, maybe something like:
- Generic datasets
- Oncology datasets
- Ophthalmology datasets
- etc...

Image

@Lina2689 @Fanny-Gautier FYI - no real rush to do this, but just an idea

### PR DESCRIPTION
**Pull request by Lina2689**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: data-raw/create_sdtms_data.R [hunk 1]
```diff
@@ -60,6 +60,42 @@ generate_hyperlink <- function(url, link_text = "") {
   paste0("[Access the source of the ", link_text, " dataset.](", url, ")")
 }
 
+# Helper function to retrieve the dataset keyword to group by TA
+#' @description Retrieve the dataset keyword to group datasets by TA.
+#' @param dataset_name The name of the dataset.
+get_dataset_keyword <- function(dataset_name) {
+  oncology <- c(
+    "tu_onco", "tu_onco_recist", "tr_onco", "tr_onco_recist", "ts",
```
[the remark this answers is not in the document]
Comment (Lina2689) [reply data-raw/create_sdtms_data.R]: no, I will move it to general category.

File: data-raw/create_sdtms_data.R [hunk 2]
```diff
@@ -60,6 +60,42 @@ generate_hyperlink <- function(url, link_text = "") {
   paste0("[Access the source of the ", link_text, " dataset.](", url, ")")
 }
 
+# Helper function to retrieve the dataset keyword to group by TA
+#' @description Retrieve the dataset keyword to group datasets by TA.
+#' @param dataset_name The name of the dataset.
+get_dataset_keyword <- function(dataset_name) {
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply data-raw/create_sdtms_data.R]: On review, my suggestion would be to add a category to `inst/extdata/sdtms-specs.json` and have this helper function simply read in that value for each dataset. If you do go down this route, it would be important to document this somewhere ("How to Update" section of the README?) in order for the process not to be too mysterious.

File: README.md [hunk 3]
```diff
@@ -54,7 +54,41 @@ Firstly, make a GitHub issue in [`{pharmaversesdtm}`](https://github.com/pharmav
...
+`data-raw/create_sdtms_data.R`
+
+This script regenerates:
+
+-   Documentation `.R` files in `R/`
+-   `.Rd` files in `man/`
+-   `Test Name`/`Test Code` table inclusion (when present)
+-   Dataset grouping by `Therapeutic Area`
+
+### When You Must Update `sdtms-specs.json`
```
Comment (manciniedoardo) [inline README.md]: this is good, i wonder though if we can do without it and simply add an item about updating/reviewing `sdtm-specs.json` in both the "Adding New SDTM Datasets" and "Updating Existing SDTM Datasets" for brevity. what do you think?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Looking great @Lina2689 - just some thoughts on writing more robust code and documentation.

File: README.md (hunk 3 above)
Comment (Lina2689) [reply README.md]: Sure, will update this section
